internal/configs/version2/nginx-plus.virtualserver.tmpl

```diff
@@ -423,6 +423,11 @@ server {
         {{ $proxyOrGRPC }}_set_header username $jwt_claim_sub;
         {{ end }}
 
+        {{ if $s.OIDC.AccessTokenEnable }}
+        {{ $proxyOrGRPC }}_set_header Authorization "Bearer $access_token";
+        {{ $proxyOrGRPC }}_intercept_errors on;
```

Comment:
do we always want to intercept errors? 
I see this comment in the repo
```
# Intercept and redirect "401 Unauthorized" proxied responses to nginx
# for processing with the error_page directive. Necessary if Access Token
# can expire before ID Token.
```
but I'm not sure if we could have side effects if it's always enabled 🤔

Reply:
- Thanks for the good question. 

- Yeah, we could think of several scenarios.
  - **use case 1**. if customers always require to login via IdP to access their proxied apps:
     - The`{{ $proxyOrGRPC }}_intercept_errors on` might help based on the N+ Ref. Implementation.
  - **use case 2**. if customers access a landing page (which isn't required the login for the first time) to show public information without OIDC flow, and if they want to access proxied APIs to show private information after they click `login` button via UI:
     - The `{{ $proxyOrGRPC }}_intercept_errors on` can be removed. Because the UI app can either just show error message or call `/login` endpoint (app level location) to start `@do_oidc_flow` from their browser.
- So, I have added an additional option (`InterceptErrorEnable`) to enable intercepting error for customers to choose their preference.

Let me know if you have other opinion.

Reply:
@lucacome I agree. I think we should go with errorPages. Since we already use `proxy_intercept_errors` I am concerned that we will generate duplicate directives of `proxy_intercept_errors` which will cause NGINX to fail.
I think safer approach is `error_pages`.

Reply:
- Thanks @lucacome and @jasonwilliams14 for your feedback.
- I have just removed the `InterceptErrorEnable` and `{{ $proxyOrGRPC }}_intercept_errors` for this PR as the following reasons:
  - There is already the same directive.
  - There is already `error_page 401 = @do_oidc_flow` that is created in the config once the OIDC is enabled.
     - If `@do_oidc_flow` is part of the `ErrorPage`, and if we remove the existing directive, then it might be inconvenient for customers.
     - If we just add `error_page 401 = @do_oidc_flow` using the `ErrorPage`, it would be duplicated config as well. 
- Therefore, I removed it for this PR as for now.
- Feel free to let me know if you have better ideas.